### ISSUE TITLE
OpenAPI specification for APIs

### ISSUE THREAD

**Issue by tanmaykm** [warrant-dev/warrant#179]
**Is your feature request related to a problem? Please describe.**

Warrant APIs are based on HTTP, well structured and mostly in REST format. It will be useful to integrate with other systems if an OpenAPI specification is provided.

**Describe the solution you'd like**

Providing an OpenAPI specification for the APIs will allow using it from a wide variety of languages. E.g. https://github.com/OpenAPITools/openapi-generator#overview

**Describe alternatives you've considered**

Not sure if this was considered. And whether it will be suitable for the APIs and useful to move the [SDKs](https://github.com/warrant-dev/warrant#sdks) to use.

**Additional context**

- none -

**Comment by akajla09** [comment]
@tanmaykm thanks for opening this for discussion. We have talked internally about adopting OpenAPI but haven't yet made a decision on it.

I'm curious - are you looking for an SDK in a language we don't currently support (e.g. .NET, Rust)?

**Comment by tanmaykm** [comment]
Thanks for the quick response @akajla09 !
Yes, we may need to access warrant from Julia and Lua, and it would be easier if we could just generate from a spec.

Unless of course if the SDKs do more than just invoking the API endpoints. In that case do you recommend using the SDKs only?

**Comment by akajla09** [comment]
Makes sense. I think more than likely we'll add an OpenAPI spec so clients can generate their own SDKs in whatever language they need. I'll provide an update here once we figure that out. In the interim, you're more than welcome to build your own simple Julia or Lua wrapper SDK as needed (there are just a handful of endpoints).
